crates/autopilot/src/decoded_settlement.rs

```diff
@@ -97,6 +98,12 @@ impl DecodedTrade {
     }
 }
 
+#[derive(Debug, PartialEq, Eq)]
+pub struct TradeWithOrderUid {
+    pub inner: DecodedTrade,
+    pub order_uid: OrderUid,
+}
```

Comment:
I tried to avoid this type by adding the order_uid right into the DecodedTrade, but that led to problems to maintain the proper logging here: https://github.com/cowprotocol/services/pull/2688/files#diff-a8d36d62f168f7d7edcae25a871749822c1d249f561b19e39fac4514f0622087R251-R256

Reply:
I think I would just make `uid` a field on `DecodedTrade` and default initialize it and then afterwards call `DecodedTrade::compute_uid()` (now renamed private function) and set the `uid` field to the computed value.
That way you can still log all the raw data when decoding fails but you don't have an awkward wrapper type.

Reply:
Too bad domain settlement code is not yet ready to be used. There, we already have order_uid: https://github.com/cowprotocol/services/blob/main/crates/autopilot/src/domain/settlement/trade.rs#L15